subprojects/core/src/main/java/org/gradle/api/internal/plugins/ImperativeOnlyPluginTarget.java

```diff
@@ -50,8 +50,8 @@ public void applyRules(@Nullable String pluginId, Class<?> clazz) {
     }
 
     @Override
-    public void applyImperativeRulesHybrid(@Nullable String pluginId, Plugin<?> plugin) {
-        applyRules(pluginId, plugin.getClass());
+    public void applyImperativeRulesHybrid(@Nullable String pluginId, Plugin<?> plugin, Class<?> clazz) {
+        applyRules(pluginId, clazz);
```

Comment:
IIUC, `clazz` is now the abstract type, right? Shall we call this `declaringClass` or something?